packages/svelte-materialify/@types/FluidGrid.d.ts

```diff
@@ -0,0 +1,40 @@
+import { SvelteComponent } from 'svelte-materialify/@types/shared';
+
+interface FluidGridProps {
+  /** whether this is a container */
+  container?: boolean;
```

Comment:
at least one of `container` or `item` has to be set to true.
can that be expressed somehow ?

Reply:
You can use [discriminated unions/algebraic data types](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#discriminated-unions) when you want to declare mutually exclusive states.

```ts
interface FluidGridSharedProps {
  // all common props go here
}

interface FluidGridContainerProps extends FluidGridSharedProps {
  container: true;
  item?: false;
}

interface FluidGridItemProps extends FluidGridSharedProps {
  item: true;
  container?: false;
}

type FluidGridProps = FluidGridContainerProps | FluidGridItemProps
```

Be aware that as a union type rather than an interface there are more limitations placed on `FluidGridProps` so this approach has drawbacks to consider. One such drawback is that you cannot extend it with a new child interface:

```ts
interface MyCustomGridProps extends FluidGridProps {
  foo: string;
}
```
^^^ this will throw a compiler error

Instead you must declare a child as a type instead
```ts
type MyCustomGridProps = FluidGridProps & {
  foo: string
}
```
^^^ this will compile

I haven't reviewed much of the code base to check if this approach is being used for this project but hopefully this answers your question.